Splashpage - Safari event countdown

jomey opened the pull request:
This is a first attempt to investigate the event countdown issues on the splashpage with the Safari browser.

Logged here: https://github.com/uwhackweek/jupyterbook-template/issues/50

jomey committed:
Splashpage - 'Defer' javascript resource loading.

This is an attempt to fix countdown issues with Safari, but also a
general better approach for loading content dependent JS.

From the MDN docs:
This Boolean attribute is set to indicate to a browser that the script
is meant to be executed after the document has been parsed, but before
firing DOMContentLoaded.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attributes

jomey committed:
Splashpage JS - Wait for the DOM ready event before execution

This wraps the countdown start and the automatic schedule day selection
to wait for the DOM to be fully parsed.

jomey committed:
Splashpage JS - Optimize event countdown a bit

Deliver the static HTML content via cookiecutter and only update the
relevant numbers in the counter. Before it was updating almost the
entire countdown every second.

jomey commented:
Successfully tested in Firefox and Chrome on Linux.

scottyhq commented:
The HTML & JS is foreign to me, But I'm seeing the same error and no countdown on Safari unfortunately. On the plus side, it's not really noticeable - the website is still very functional and clean looking

jomey committed:
Splashpage - Use precise event time for countdown

Use the exact time of the first session for the event countdown.

jomey commented:
> The HTML & JS is foreign to me, But I'm seeing the same error and no countdown on Safari unfortunately. On the plus side, it's not really noticeable - the website is still very functional and clean looking

The error message was not the cause of the issue. It basically came down to the difference of Javascript date parsing between Safari, Chrome, and Firefox. 

I changed the configuration of the timer to be an explicit item in the `cookiecutter.yaml`. Tested this just again successfully on the same platforms, plus Safari and Firefox on my iPad and I can see the countdown now.

Added bonus is that we are counting down exactly to 8:30 PST 😄

scottyhq reviewed (approved):
Neat! Thanks @jomey
